fvm/context.go

```diff
@@ -174,6 +178,14 @@ func WithAccountFreezeAvailable(accountFreezeAvailable bool) Option {
 	}
 }
 
+// WithServiceEventCollectionEnabled enables service event collection
+func WithServiceEventCollectionEnabled() Option {
+	return func(ctx Context) Context {
+		ctx.ServiceEventCollectionEnabled = true
+		return ctx
+	}
+}
```

Comment:
This is a security measure that we discussed recently with Jordan, only during the execution of the system chunk we expect service events, otherwise, we shouldn't let it happen.